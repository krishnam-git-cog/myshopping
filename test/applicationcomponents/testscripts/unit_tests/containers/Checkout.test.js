import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Checkout from '../../../../../src/containers/checkout/Checkout';
import { createMemoryHistory } from 'history';

describe('Checkout test', () => {

    let shallow;
    beforeEach(() => {
        shallow = createShallow();
    });

    it('Checkout should match snapshot', () => {
        const history = createMemoryHistory('/')
        const component = shallow(<Checkout history={history} />);
        expect(component).toMatchSnapshot();
    });

    it('Checkout error message', () => {
        const history = createMemoryHistory('/')
        const component = shallow(<Checkout history={history} />);
        component.find('#buttonnext').simulate('click');
        expect(component.state('showdaerror')).toBe(true);
    });

    it('Checkout valid data', () => {
        const history = createMemoryHistory('/')
        const component = shallow(<Checkout history={history} />);
        component.setState({ daisvalid: true});
        component.find('#buttonnext').simulate('click');
        expect(component.state('daisvalid')).toBe(true);
    });

    it('Checkout back button', () => {
        const history = createMemoryHistory('/')
        const component = shallow(<Checkout history={history} />);
        component.find('#backbutton').simulate('click');
        expect(component.state('activeStep')).toBe(-1);
    });
});