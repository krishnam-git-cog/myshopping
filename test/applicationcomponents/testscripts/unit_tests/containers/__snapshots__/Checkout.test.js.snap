// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout test Checkout should match snapshot 1`] = `ShallowWrapper {}`;
